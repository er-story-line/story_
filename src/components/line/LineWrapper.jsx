import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const LineWrapper = styled.div`
  background: ${props => props.backgroundColor};
  height: 100%;
`

const wrapper = ({ children, styles }) => (
  <LineWrapper {...styles}>{children}</LineWrapper>
)

wrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  styles: PropTypes.shape({
    backgroundColor: PropTypes.string,
  }),
}

wrapper.defaultProps = () => ({
  children: null,
  styles: {
    backgroundColor: 'white',
  },
})

export default wrapper
