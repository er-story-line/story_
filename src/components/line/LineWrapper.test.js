import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import LineWrapper from './LineWrapper'

describe('LineWrapper', () => {
  const styles = {
    backgroundColor: 'white',
  }
  it('renders', () => {
    const tree = renderer.create(<LineWrapper {...styles} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
