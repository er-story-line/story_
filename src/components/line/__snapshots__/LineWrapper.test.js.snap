// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineWrapper renders 1`] = `
.c0 {
  margin-top: 100px;
  height: 100%;
}

<div
  className="c0"
/>
`;
